warehouse/integrations/bigquery/bigquery.go

```diff
@@ -443,41 +441,41 @@ func (bq *BigQuery) loadTable(ctx context.Context, tableName string) (
 	)
 	log.Infon("started loading")
 
-	loadFileLocations, err := bq.loadFileLocations(ctx, tableName)
+	loadFileLocation, err := bq.loadFileLocation(ctx, tableName)
 	if err != nil {
-		return nil, nil, fmt.Errorf("getting load file locations: %w", err)
+		return nil, nil, fmt.Errorf("getting load file location: %w", err)
 	}
 
-	gcsRef := bigquery.NewGCSReference(warehouseutils.GetGCSLocations(
-		loadFileLocations,
-		warehouseutils.GCSLocationOptions{},
-	)...)
+	gcsRef := bigquery.NewGCSReference(loadFileLocation)
 	gcsRef.SourceFormat = bigquery.JSON
 	gcsRef.MaxBadRecords = 0
 	gcsRef.IgnoreUnknownValues = false
 
 	return bq.loadTableByAppend(ctx, tableName, gcsRef, log)
 }
 
-func (bq *BigQuery) loadFileLocations(
+func (bq *BigQuery) loadFileLocation(
 	ctx context.Context,
 	tableName string,
-) ([]warehouseutils.LoadFile, error) {
+) (string, error) {
 	switch tableName {
 	case warehouseutils.IdentityMappingsTable, warehouseutils.IdentityMergeRulesTable:
 		loadfile, err := bq.uploader.GetSingleLoadFile(
 			ctx,
 			tableName,
 		)
 		if err != nil {
-			return nil, fmt.Errorf("getting single load file for table %s: %w", tableName, err)
+			return "", fmt.Errorf("getting single load file for table %s: %w", tableName, err)
 		}
-		return []warehouseutils.LoadFile{loadfile}, nil
+		return loadfile.Location, nil
```

Comment:
It looks to me that in case of `IdentityMappingsTable` we are returning the file location and the folder location in `default` case. Isn't this inconsistent ?

Reply:
Yeah, ID Resolution is some legacy code where load files are not generated on slave instead of master and it does not follow a proper naming convention.

Reply:
Also, ID Resolution is only for BQ and Snowflake.